aten/src/ATen/functorch/BatchedTensorImpl.cpp

```diff
@@ -15,7 +15,11 @@ namespace functorch {
 
 BatchedTensorImpl::BatchedTensorImpl(DispatchKeySet key_set, Tensor value, int64_t bdim, int64_t level)
   : TensorImpl(
-      key_set.add(DispatchKey::FuncTorchBatched),
+      key_set.add(
+          value.is_nested() ? DispatchKeySet({
+              DispatchKey::NestedTensor,
+              DispatchKey::BatchedNestedTensor
+          }) : DispatchKeySet({DispatchKey::FuncTorchBatched})),
```

Comment:
Definitely have had to deal with the mixed batched NT / T stuff some. My initial thoughts are that we may want to beef up the existing fallback to handle both NT and T cases generically. It's easy to imagine a function that takes in only a normal tensor input and returns a normal tensor of size varying based on the input. In this case, I think we'd want support for NT outputs. 

side note: does it make sense to you to have a `BatchedTensorImpl` that wraps a NT defining the `NestedTensor` key? I added this so that `is_nested() -> True` for these but this may not be a good enough reason.